lib/actions/user.js

```diff
@@ -55,19 +63,21 @@ export function fetchOrInitializeUser (auth) {
       // }
       // TODO: Improve AWS response.
 
-      const resultData = result.data
-      const isNewAccount = result.status === 'error' || (resultData && resultData.result === 'ERR')
-
+      const isNewAccount = fetchUserStatus === 'error' || (user && user.result === 'ERR')
       if (!isNewAccount) {
         // TODO: Move next line somewhere else.
-        if (resultData.savedLocations === null) resultData.savedLocations = []
-        dispatch(setCurrentUser({ accessToken, user: resultData }))
+        if (user.savedLocations === null) user.savedLocations = []
```

Comment:
This seems like something that should be fixed on the server side (i.e., initializing user's saved locations to an empty arraylist).

Reply:
I don't know... We could also replace this with a null check where saved locations are rendere.

Reply:
Addressed in https://github.com/opentripplanner/otp-react-redux/pull/174/commits/277d868571d617c7be128c0ca73c2ede622ff465. I added the null checks where needed.

Reply:
@binh-dam-ibigroup, I don't think that's the right approach. I really just think it should be initialized to an empty list on the server side. Please just add a `fixme` here and let's create an issue in otp-middleware for this. (adding multiple null checks in other places is just spreading the problem even further.

Reply:
Noted in https://github.com/ibi-group/otp-middleware/issues/57 and https://github.com/opentripplanner/otp-react-redux/pull/174/commits/982a8777c8b36e2392de2ee5348bd5ee4d0a140d.